internal/era2/era.go

```diff
@@ -0,0 +1,443 @@
+	return td, nil
+}
+
+// retrieves the raw body frame in bytes of a specific block
+func (e *Era) GetRawBodyFrameByNumber(blockNum uint64) ([]byte, error) {
+	off, err := e.bodyOff(blockNum)
+	if err != nil {
+		return nil, err
+	}
+	r, _, err := e.s.ReaderAt(TypeCompressedBody, int64(off))
+	if err != nil {
+		return nil, err
+	}
+	return io.ReadAll(r)
+}
+
+// retrieves the raw receipts frame in bytes of a specific block
+func (e *Era) GetRawReceiptsFrameByNumber(blockNum uint64) ([]byte, error) {
+	off, err := e.rcptOff(blockNum)
+	if err != nil {
+		return nil, err
+	}
+	r, _, err := e.s.ReaderAt(TypeCompressedReceipts, int64(off))
+	if err != nil {
+		return nil, err
+	}
+	return io.ReadAll(r)
+}
+
+// retrieves the raw proof frame in bytes of a specific block proof
+func (e *Era) GetRawProofFrameByNumber(blockNum uint64) ([]byte, error) {
+	off, err := e.proofOff(blockNum)
+	if err != nil {
+		return nil, err
+	}
+	r, _, err := e.s.ReaderAt(TypeProof, int64(off))
+	if err != nil {
+		return nil, err
+	}
+	return io.ReadAll(r)
+}
+
+// loads in the index table containing all offsets and caches it
+func (e *Era) loadIndex() error {
+	var err error
+	e.m.length, err = e.f.Seek(0, io.SeekEnd)
+	if err != nil {
+		return err
+	}
+
+	b := make([]byte, 16)
+	if _, err = e.f.ReadAt(b, e.m.length-16); err != nil {
+		return err
+	}
+	e.m.components = binary.LittleEndian.Uint64(b[0:8])
+	e.m.count = binary.LittleEndian.Uint64(b[8:16])
+
+	payloadlen := 8 + 8*e.m.count*e.m.components + 16 // 8 for start block, 8 per property per block, 16 for the number of properties and the number of blocks
+	tlvstart := e.m.length - int64(payloadlen) - 8
+	e.indstart = tlvstart
+
+	_, err = e.f.ReadAt(b[:8], tlvstart+8)
+	if err != nil {
+		return err
+	}
+
+	e.m.start = binary.LittleEndian.Uint64(b[:8])
+	num := int(e.m.components)
+
+	totaloffsets := num * int(e.m.count)
+	offBytes := make([]byte, totaloffsets*8)
+	offsetArea := tlvstart + 8 + 8 // 8 for the header size, 8 for the start number
+	_, err = e.f.ReadAt(offBytes, offsetArea)
+	if err != nil {
+		return err
+	}
+
+	var off int64 = 0 // start at byte-0 of file
+
+	for off < e.indstart { // never enter the Block-Index TLV
+		typ, length, err := e.s.ReadMetadataAt(off)
+		if err != nil {
+			return fmt.Errorf("error reading TLV at offset %d: %w", off, err)
+		}
+		if typ == TypeAccumulatorRoot {
+			e.rootheader = uint64(off) // remember absolute header offset
+			break                      // found
+		}
+		off += 8 + int64(length) // headersize plus length to jump to next TLV header
+	}
+	return nil
+}
+
+// Getter methods to calculate offset of a specific component in the file.
+func (e *Era) headerOff(num uint64) (uint64, error) { return e.indexOffset(num, compHeader) }
+func (e *Era) bodyOff(num uint64) (uint64, error)   { return e.indexOffset(num, compBody) }
+func (e *Era) rcptOff(num uint64) (uint64, error)   { return e.indexOffset(num, compReceipts) }
```

Comment:
```suggestion
func (e *Era) receiptOff(num uint64) (uint64, error)   { return e.indexOffset(num, compReceipts) }
```
Really no reason to ever abbreviate by taking out vowels in golang.